pkg/scheduler/framework/plugins/clusteraffinity/azure/compute.go

```diff
@@ -0,0 +1,131 @@
+	"go.goms.io/fleet/pkg/clients/azure/compute"
+	"go.goms.io/fleet/pkg/propertyprovider/azure"
+)
+
+const (
+	// AzureLocationLabelKey is the label key for Azure location
+	AzureLocationLabelKey = "fleet.azure.com/location"
+
+	// AzureSubscriptionIDLabelKey is the label key for Azure subscription ID
+	AzureSubscriptionIDLabelKey = "fleet.azure.com/subscription-id"
+)
+
+// ComputeService is the  implementation of ComputeService.
+type ComputeService struct {
+	// ComputeClient is the Azure compute client used to validate Compute requirements.
+	ComputeClient compute.AttributeBasedVMSizeRecommenderClient
+}
+
+// NewComputeService creates a new Azure Compute service
+func NewComputeService(computeClient compute.AttributeBasedVMSizeRecommenderClient) *ComputeService {
+	return &ComputeService{
+		ComputeClient: computeClient,
+	}
+}
+
+// ValidateComputeRequirement validates a Compute requirement against Azure APIs.
+func (s *ComputeService) ValidateComputeRequirement(
+	cluster *clusterv1beta1.MemberCluster,
+	req placementv1beta1.PropertySelectorRequirement,
+) (bool, error) {
+	subID, location := extractAzureInfoFromLabels(cluster)
+	if subID == "" || location == "" {
+		return false, fmt.Errorf("cluster %s does not have required Azure labels", cluster.Name)
+	}
+	capacity, sku, err := extractCapacityRequirements(req)
+	if err != nil {
+		return false, fmt.Errorf("failed to extract capacity requirement: %w", err)
+	}
+
+	request := &computev1.GenerateAttributeBasedRecommendationsRequest{
+		SubscriptionId: subID,
+		Location:       location,
+		RegularPriorityProfile: &computev1.RegularPriorityProfile{
+			CapacityUnitType: computev1.CapacityUnitType_CAPACITY_UNIT_TYPE_VM_INSTANCE_COUNT,
+			TargetCapacity:   uint32(capacity.Value()), //nolint:gosec // safe: capacity is derived from user input should be relatively small
+		},
+		ResourceProperties: &computev1.ResourceProperties{
+			VmAttributes: &computev1.VMAttributes{
+				AllowedVmSizes: []string{sku},
+			},
+		},
+		RecommendationProperties: &computev1.RecommendationProperties{
+			RestrictionsFilter: computev1.RecommendationProperties_RESTRICTIONS_FILTER_QUOTA_AND_OFFER_RESTRICTIONS,
+		},
+	}
+
+	respObj, err := s.ComputeClient.GenerateAttributeBasedRecommendations(context.Background(), request)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate VM size recommendations from Azure: %w", err)
+	}
+
+	available := false
+	for _, vm := range respObj.RecommendedVmSizes.RegularVmSizes {
+		if vm.Name == sku {
+			available = true
+			klog.V(2).Infof("SKU %s is available in cluster %s", sku, cluster.Name)
+			break
+		}
+	}
+
+	return available, nil
+}
+
+// extractAzureInfoFromLabels extracts subscription ID and location from MemberCluster labels.
+// Returns the subscription ID and location if found, empty strings otherwise.
+func extractAzureInfoFromLabels(cluster *clusterv1beta1.MemberCluster) (subscriptionID, location string) {
```

Comment:
move it to pkg/utils/labels/labels.go and make it public and generic.

for example,

GetLabelFromMemberCluster
(cluster *clusterv1beta1.MemberCluster, key string) string